iface/interface.go

```diff
@@ -31,39 +31,61 @@ import (
 
 	log "github.com/Sirupsen/logrus"
 
-	"github.com/intelsdi-x/snap-plugin-utilities/ns"
-	"github.com/intelsdi-x/snap-plugin-utilities/str"
 	"github.com/intelsdi-x/snap/control/plugin"
 	"github.com/intelsdi-x/snap/control/plugin/cpolicy"
 	"github.com/intelsdi-x/snap/core"
+	"github.com/intelsdi-x/snap/core/ctypes"
 	"github.com/intelsdi-x/snap/core/serror"
+
+	"github.com/intelsdi-x/snap-plugin-utilities/ns"
+	"github.com/intelsdi-x/snap-plugin-utilities/str"
 )
 
 const (
-	// VENDOR namespace part
-	VENDOR = "intel"
-	// FS namespace part
-	FS = "procfs"
-	// PLUGIN name namespace part
-	PLUGIN = "iface"
-	// VERSION of interface info plugin
-	VERSION = 3
+	// Name of plugin
+	PluginName = "iface"
+	// Version of plugin
+	PluginVersion = 3
+	// Type of plugin
+	pluginType = plugin.CollectorPluginType
+
+	nsVendor = "intel"
+	nsClass  = "procfs"
+	nsType   = "iface"
 )
 
-var ifaceInfo = "/proc/net/dev"
+// prefix in metric namespace
+var prefix = []string{nsVendor, nsClass, nsType}
+
+// added init PID 1 so that we take the global namespace
+// to be sure to gather all possible interfaces wherever
+// the plugin gets executed
+var ifaceInfo = "/proc/1/net/dev"
+
+// Meta returns plugin meta data
+func Meta() *plugin.PluginMeta {
+	return plugin.NewPluginMeta(
+		PluginName,
+		PluginVersion,
+		pluginType,
+		[]string{},
+		[]string{plugin.SnapGOBContentType},
+		plugin.ConcurrencyCount(1),
+	)
+}
 
 // GetMetricTypes returns list of available metric types
 // It returns error in case retrieval was not successful
-func (iface *ifacePlugin) GetMetricTypes(_ plugin.ConfigType) ([]plugin.MetricType, error) {
+func (iface *ifacePlugin) GetMetricTypes(cfg plugin.ConfigType) ([]plugin.MetricType, error) {
 	metricTypes := []plugin.MetricType{}
 
-	if err := getStats(iface.stats); err != nil {
+	if err := getStats(cfg.Table(), iface.stats); err != nil {
```

Comment:
What do you think about using `config.GetConfigItem(cfg, "proc_path")` before calling `getStats` and pass proper path to it?